Convert action to composite

knightdave opened the pull request:
Convert to composite action. Close https://github.com/insightsengineering/thevalidatoR/issues/29.

dinakar29 commented on action.yml:
```diff
@@ -19,12 +19,71 @@ inputs:
+    - name: Set R Library home on Linux
+      
+      run: |
+        mkdir -p /__w/_temp/Library
```
Isn't this what Github Action uses on the runner's host machine? I'd be wary of putting anything there an using `$GITHUB_WORKSPACE` instead

knightdave replied:
It's because of this how action run inside docker container. It [spawns container with](https://github.com/insightsengineering/thevalidatoR/runs/4311717111?check_suite_focus=true#step:2:63):

```
 /usr/bin/docker create --name b8ad9de4427c4411bb5cb3bf84e7cbcc_rockerverse411_9885f6 --label e28490 --workdir /__w/thevalidatoR/thevalidatoR --network github_network_b75ab2503ef34c109dd2b5f9d9449583  -e "HOME=/github/home" -e GITHUB_ACTIONS=true -e CI=true -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/home/runner/work":"/__w" -v "/home/runner/runners/2.284.0/externals":"/__e":ro -v "/home/runner/work/_temp":"/__w/_temp" -v "/home/runner/work/_actions":"/__w/_actions" -v "/opt/hostedtoolcache":"/__t" -v "/home/runner/work/_temp/_github_home":"/github/home" -v "/home/runner/work/_temp/_github_workflow":"/github/workflow" --entrypoint "tail" rocker/verse:4.1.1 "-f" "/dev/null"
```

It means inside container we have this `/home/runner/work/_temp` mounted as `/__w/_temp`.
`run` step is running inside container so we need to rely on this weird path. The variable `GITHUB_WORKSPACE` refers to the working path in machine - so it will be `/home/runner/work/org/repo` [doc](https://docs.github.com/en/actions/learn-github-actions/environment-variables). The path `/home/runner/work/_temp` will be the path that we want to cache with cache action.

Basically it's a workaround to properly use `_temp` directory for caching inside docker container - I can search for article where I found this.

knightdave replied:
I updated this to at least use env vars where I can

dinakar29 replied:
Ah good. Thank you!

dinakar29 commented on action.yml:
```diff
@@ -19,12 +19,71 @@ inputs:
+        export TEXMFVAR=/__w/_temp/texlive/texmf-var
+        tlmgr init-usertree
+        echo "::set-output name=TEX_LIVE_VERSION::$(tlmgr --version |tail -1 |awk '{print $NF}')"
+      shell: bash
```
Might be good to install TinyTex instead of relying on a TexLive installation.

knightdave replied:
done

dinakar29 commented on dependencies.R:
```diff
@@ -11,7 +11,33 @@ github_packages <- c(
+cp -r ~/.TinyTeX/. ${RUNNER_TEMP}/TinyTeX
+rm -rf ~/.TinyTeX
+${RUNNER_TEMP}/TinyTeX/bin/*/tlmgr path add
+tlmgr install latex-bin luatex xetex ae bibtex context inconsolata listings makeindex metafont mfware parskip pdfcrop tex tools url xkeyval courier ec
```
Ideally I'd like this to be an input to the action - i.e. add additional TeX packages if they are missing or needed by a package. 
Let's keep these packages as defaults, but let's also provide the ability to add extra pacakges.

knightdave replied:
the whole list without `courier` and `ec` comes from rocker/verse - they are installed by default.
I added parameter that allows install more of them then only those two.

knightdave commented:
Finally tmgr caching working:

No cache:
![Screenshot 2021-11-24 at 16 31 49](https://user-images.githubusercontent.com/13086223/143268740-a9b5dd18-367f-40d4-8ff8-e60a69aaa29a.png)


With cache:
![Screenshot 2021-11-24 at 16 37 22](https://user-images.githubusercontent.com/13086223/143268807-ed2575f0-e92f-4fc0-84e5-3319b4b3428a.png)

(aragog 7m --> 2m decreased time)

knightdave commented:
@dinakar29  - do you want me to add something more? Is it OK to merge?

dinakar29 commented:
> @dinakar29 - do you want me to add something more? Is it OK to merge?

Can we add this? https://github.com/insightsengineering/thevalidatoR/pull/45/files#r756155629

knightdave commented:
> Can we add this? [#45 (comment)](https://github.com/insightsengineering/thevalidatoR/pull/45#discussion_r756155629)

I added `additional_tlmgr_packages` parameter and just right now updated tests.

I'll wait then for your PR.

dinakar29 commented:
Merging this. Will release this as v1.1
